openedx/features/genplus_features/genplus/rmunify.py

```diff
@@ -18,19 +22,15 @@ class RmUnifyException(BaseException):
     pass
 
 
-class RmUnify:
-    ORGANISATION = 'organisation/'
-    TEACHING_GROUP = '{}{}/teachinggroup/'
-    REGISTRATION_GROUP = '{}{}/registrationgroup/'
-
+class BaseRmUnify:
     def __init__(self):
         self.key = settings.RM_UNIFY_KEY
         self.secret = settings.RM_UNIFY_SECRET
         self.timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
 
-    def fetch(self, source, source_id=None):
-        headers = {"Authorization": "Unify " + self.key + "_" + self.timestamp + ":" + self.hashed}
-        url = self.generate_url(source, source_id)
+    def fetch(self, source, source_id=None, provisioning=False):
+        headers = self.get_header()
+        url = self.generate_url(source, source_id, provisioning=provisioning)
```

Comment:
Can we remove provisioning code from BaseRmUnify and override fetch function in RmUnifyProvisioning class?